src/sentry/tasks/groupowner.py

```diff
@@ -20,84 +23,98 @@
-                ).order_by("-date_added")
-                if not owners.exists():
+                if cache.get(cache_key):
+                    # Only process once per OWNER_CACHE_LIFE seconds.
                     metrics.incr(
-                        "sentry.tasks.process_suspect_commits.aborted",
-                        tags={"detail": "maxed_owners_none_old"},
+                        "sentry.tasks.process_suspect_commits.skipped",
+                        tags={"detail": "too_many_owners"},
                     )
                     can_process = False
+                else:
+                    project = Project.objects.get_from_cache(id=project_id)
+                    owners = GroupOwner.objects.filter(
+                        group_id=group_id,
+                        project=project,
+                        organization_id=project.organization_id,
+                        type=GroupOwnerType.SUSPECT_COMMIT.value,
+                    )
+                    owner_count = owners.count()
+                    if owner_count >= PREFERRED_GROUP_OWNERS:
+                        owners = owners.filter(
+                            date_added__lte=timezone.now() - PREFERRED_GROUP_OWNER_AGE
+                        ).order_by("-date_added")
+                        if not owners.exists():
+                            metrics.incr(
+                                "sentry.tasks.process_suspect_commits.aborted",
+                                tags={"detail": "maxed_owners_none_old"},
+                            )
+                            can_process = False
 
-            cache.set(cache_key, True, OWNER_CACHE_LIFE)
+                    cache.set(cache_key, True, OWNER_CACHE_LIFE)
 
-        if can_process:
-            with metrics.timer("sentry.tasks.process_suspect_commits.process_loop"):
-                metrics.incr("sentry.tasks.process_suspect_commits.calculated")
-                try:
-                    with metrics.timer(
-                        "sentry.tasks.process_suspect_commits.get_serialized_event_file_committers"
-                    ):
-                        committers = get_event_file_committers(project, event)
-                    owner_scores = {}
-                    for committer in committers:
-                        if "id" in committer["author"]:
-                            author_id = committer["author"]["id"]
-                            for commit, score in committer["commits"]:
-                                if score >= MIN_COMMIT_SCORE:
-                                    owner_scores[author_id] = max(
-                                        score, owner_scores.get(author_id, 0)
-                                    )
+                if can_process:
+                    with metrics.timer("sentry.tasks.process_suspect_commits.process_loop"):
+                        metrics.incr("sentry.tasks.process_suspect_commits.calculated")
```

Comment:
Nit: I realised this is unnecessary since we can just get the counts of `process_loop`